velox/dwio/parquet/tests/duckdb_reader/ParquetTableScanTest.cpp

```diff
@@ -227,6 +252,51 @@ TEST_F(ParquetTableScanTest, countStar) {
   assertQuery(plan, {split}, "SELECT 20");
 }
 
+class NativeParquetTableScanTest
+    : public ParquetTableScanTest,
+      public testing::WithParamInterface<ParquetReaderType> {
+ public:
+  NativeParquetTableScanTest() : ParquetTableScanTest(GetParam()) {}
+};
+
+TEST_P(NativeParquetTableScanTest, decimalSubfieldFilter) {
+  // decimal.parquet holds two columns (a: DECIMAL(5, 2), b: DECIMAL(20, 5)) and
+  // 20 rows (10 rows per group). Data is in plain uncompressed format:
+  //   a: [100.01 .. 100.20]
+  //   b: [100000000000000.00001 .. 100000000000000.00020]
+  std::vector<int64_t> unscaledShortValues(20);
+  std::iota(unscaledShortValues.begin(), unscaledShortValues.end(), 10001);
+  loadData(
+      getExampleFilePath("decimal.parquet"),
+      ROW({"a"}, {DECIMAL(5, 2)}),
+      makeRowVector(
+          {"a"},
+          {
+              makeShortDecimalFlatVector(unscaledShortValues, DECIMAL(5, 2)),
+          }));
+
+  assertSelectWithFilter(
+      {"a"}, {"a < 100.07"}, "", "SELECT a FROM tmp WHERE a < 100.07");
+  assertSelectWithFilter(
+      {"a"}, {"a <= 100.07"}, "", "SELECT a FROM tmp WHERE a <= 100.07");
+  assertSelectWithFilter(
+      {"a"}, {"a > 100.07"}, "", "SELECT a FROM tmp WHERE a > 100.07");
+  assertSelectWithFilter(
+      {"a"}, {"a >= 100.07"}, "", "SELECT a FROM tmp WHERE a >= 100.07");
+  assertSelectWithFilter(
+      {"a"}, {"a = 100.07"}, "", "SELECT a FROM tmp WHERE a = 100.07");
+  assertSelectWithFilter(
+      {"a"},
+      {"a BETWEEN 100.07 AND 100.12"},
+      "",
+      "SELECT a FROM tmp WHERE a BETWEEN 100.07 AND 100.12");
+}
+
+VELOX_INSTANTIATE_TEST_SUITE_P(
```

Comment:
It is a bit odd to create a parameterized test for testing only a single parameter. Should ParquetReaderType::DUCKDB also be added here ?